RecoVertex/GaussianSumVertexFit/test/TrackMix.cc

```diff
@@ -1,37 +1,65 @@
+*/
+
+// system include files
+#include <memory>
+#include <iostream>
+
+// user include files
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/GsfTrackReco/interface/GsfTrack.h"
+#include "DataFormats/GsfTrackReco/interface/GsfTrackFwd.h"
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/TrackReco/interface/TrackFwd.h"
 #include "DataFormats/VertexReco/interface/Vertex.h"
 #include "DataFormats/VertexReco/interface/VertexFwd.h"
-#include "DataFormats/TrackReco/interface/Track.h"
-#include "DataFormats/Common/interface/Handle.h"
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
 #include "FWCore/Framework/interface/MakerMacros.h"
 #include "FWCore/MessageLogger/interface/MessageLogger.h"
-#include "FWCore/Framework/interface/ESHandle.h"
-#include "DataFormats/GsfTrackReco/interface/GsfTrack.h"
-#include "DataFormats/GsfTrackReco/interface/GsfTrackFwd.h"
-
-#include "TrackingTools/TransientTrack/interface/TransientTrackBuilder.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "RecoVertex/KalmanVertexFit/interface/KalmanVertexFitter.h"
+#include "RecoVertex/VertexPrimitives/interface/TransientVertex.h"
 #include "TrackingTools/Records/interface/TransientTrackRecord.h"
 #include "TrackingTools/TransientTrack/interface/TransientTrack.h"
-#include "RecoVertex/VertexPrimitives/interface/TransientVertex.h"
-#include "RecoVertex/KalmanVertexFit/interface/KalmanVertexFitter.h"
+#include "TrackingTools/TransientTrack/interface/TransientTrackBuilder.h"
 
-#include <iostream>
+// ROOT includes
+#include <TFile.h>
+
+/**
+   * This is a very simple test analyzer mean to test the KalmanVertexFitter
+   */
+
+class TrackMix : public edm::one::EDAnalyzer<> {
+public:
+  explicit TrackMix(const edm::ParameterSet&);
+  ~TrackMix();
+
+  virtual void analyze(const edm::Event&, const edm::EventSetup&);
+
+private:
+  const edm::ESGetToken<TransientTrackBuilder, TransientTrackRecord> estoken_ttk;
+  edm::EDGetTokenT<edm::View<reco::Track> > token_gsf, token_ckf;
+};
 
 using namespace reco;
 using namespace edm;
 using namespace std;
 
-TrackMix::TrackMix(const edm::ParameterSet& iConfig) : theConfig(iConfig) {
+TrackMix::TrackMix(const edm::ParameterSet& iConfig)
+    : estoken_ttk(esConsumes(edm::ESInputTag("", "TransientTrackBuilder"))) {
   token_gsf = consumes<edm::View<reco::Track> >(iConfig.getParameter<string>("gsfTrackLabel"));
   token_ckf = consumes<edm::View<reco::Track> >(iConfig.getParameter<string>("ckfTrackLabel"));
 }
 
-TrackMix::~TrackMix() {}
-
-void TrackMix::beginJob() {}
-
-void TrackMix::endJob() {}
+TrackMix::~TrackMix() = default;
 
 void TrackMix::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
   try {
```

Comment:
Could you please move to `cms::Exception`? Also, there are a few `cout` that can be changed to `edm::LogPrint`

Reply:
done in the last push.